monad-eth-testutil/examples/txgen/shared/erc20.rs

```diff
@@ -41,6 +41,50 @@ pub struct ERC20 {
     pub addr: Address,
 }
 
+#[derive(Debug, Clone)]
+pub struct ERC20Pool {
```

Comment:
This abstraction should not exist.
You should have a `erc20s: Vec<ERC20>` in the generator struct (or wherever this is used), then call self.erc20s.iter() instead, or take a slice reference